docs/create_manual.py

```diff
@@ -0,0 +1,138 @@
+https://www.sphinx-doc.org/en/master/man/sphinx-build.html
+https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html
+"""
+
+import os
+from typing import Sequence
+import subprocess
+import shutil
+import sys
+
+DOCS_DIR = os.getcwd()
+SPHINX_DIR = os.path.join(DOCS_DIR, 'sphinx')
+SPHINX_MOD_DIR = os.path.join(SPHINX_DIR, 'modules')
+SPHINX_SCRIPTS_DIR = os.path.join(SPHINX_DIR, '_scripts')
+CHIPSEC_DIR = os.path.normpath(DOCS_DIR + os.sep + os.pardir)
+
+NotWantedFilesList = [ 
+    'chipsec_tools.rst',
+    'chipsec_tools.windows.rst',
+    'setup.rst',
+    'chipsec.rst',
+    'chipsec.banner.rst',
+    'chipsec.cfg.rst',
+    'chipsec.chipset.rst',
+    'chipsec.command.rst',
+    'chipsec.defines.rst',
+    'chipsec.exceptions.rst',
+    'chipsec.file.rst',
+    'chipsec.logger.rst',
+    'chipsec.module.rst',
+    'chipsec.module_common.rst',
+    'chipsec.result_deltas.rst',
+    'chipsec_main.rst',
+    'chipsec_util.rst']
+
+def RunAutoDoc() -> None:
+    try:
+        subprocess.call(f'sphinx-apidoc -e -f -T -d 10 -o modules {CHIPSEC_DIR} {CHIPSEC_DIR}\\*test*')
+    except:
+        print('Unable to run sphinx-apidoc')
+
+def CleanupFilesNotWantedInDoc() -> None:
+    for file in NotWantedFilesList:
+        try:
+            os.remove(os.path.join(SPHINX_MOD_DIR, file))
+        except:
+            print(f'Unable to remove {file}')
+
+def RunScripts() -> None:
+    for script in os.listdir(os.path.join(SPHINX_SCRIPTS_DIR)):
+        try:
+            subprocess.call(f'python {SPHINX_SCRIPTS_DIR}\{script}')
+        except:
+            print(f'Unable to run script: {script}')
+
+def GeneratePDF() -> None:
+    try:
+        subprocess.call(f'sphinx-build -b pdf -T {SPHINX_DIR} {CHIPSEC_DIR}')
+    except:
+        print('Unable to generate PDF')
+
+def GenerateHTML() -> None:
+    try:
+        subprocess.call(f'sphinx-build -b html -T {SPHINX_DIR} {CHIPSEC_DIR}\\manual')
+    except:
+        print('Unable to generate HTML')
+
+def GenerateJSON() -> None:
+    try:
+        subprocess.call(f'sphinx-build -b json -T {SPHINX_DIR} {CHIPSEC_DIR}\\manualJson')
+    except:
+        print('Unable to generate JSON')
+
+format_options_functions = {
+    'html': GenerateHTML,
+    'json': GenerateJSON
+}
+
+def GenerateHTMLorJSON(option: str) -> None:
+    try:
+        format_options_functions[option]()
+    except:
+        print('Invalid format option')
+
+def DeleteSphinxCollateral() -> None:
+    shutil.rmtree(os.path.join(CHIPSEC_DIR, '.doctrees'))
+    shutil.rmtree(os.path.join(SPHINX_DIR, 'logs'))
+    shutil.rmtree(SPHINX_MOD_DIR)
+
+def main(argv: Sequence[str] = sys.argv[1:]):
+    print('******************** BUILDING DOCUMENTATION **************************')
+    os.chdir(SPHINX_DIR)
+    RunAutoDoc()
+    CleanupFilesNotWantedInDoc()
+    RunScripts()
+    if (len(argv)==1):
+        GenerateHTMLorJSON(argv[0])
+    GeneratePDF()
+    DeleteSphinxCollateral()
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
Should `main()` return something to indicate success/failure if this is used in automation?